pkg/payload/task_graph.go

```diff
@@ -581,6 +581,7 @@ func RunGraph(ctx context.Context, graph *TaskGraph, maxParallelism int, fn func
 				select {
 				case <-nestedCtx.Done():
 					klog.V(4).Infof("Canceled worker %d", job)
+					errCh <-nestedCtx.Err()
```

Comment:
It would be more correct to, at the end of the function, if the context is cancelled, do

```
if err := ctx.Err(); err != nil && len(errs) == 0 {
  errs = append(errs, err)
}
```

Reply:
We don't need one error per worker.

Reply:
Fixed in 40b7e58 -> 5eaad22.